Battery_Checker.c

```diff
@@ -0,0 +1,46 @@
+#include <stdio.h>
+#include "Battery_Checker.h"
+
+int IsbatteryTempCheck(float Temprature)
+{
+  if(Temprature > Min_Temp && Temprature < Max_Temp) 
+  {
+      printf("Temperature out of range!\n");
+      return Not_Ok;
+  }
+  else
+    return Ok;
+}
+
+int IsbatterySocCheck(float Soc)
+{
+  if(Soc > Min_Soc && Soc < Max_Soc) 
+  {
+    printf("State of Charge out of range!\n");
+    return Not_Ok;
+  }
+  else
+    return Ok;
+  
+} 
+
+int IsbatteryChargeRateCheck(float ChargeRate)
+{
+   if(ChargeRate > Max_ChargeRate) 
+   {
+     printf("Charge Rate out of range!\n");
+      return Not_Ok;
+   }
+  else
+     return Ok;
+}
+int IsbatteryOk(float Temperature, float Soc, float ChargeRate) 
+{
+   return IsbatteryTempCheck(Temperature) && IsbatterySocCheck(Soc) && IsbatteryChargeRateCheck(ChargeRate);
+ 
+ }
+
+int main() {
+  assert(IsbatteryOk(40, 55, 0.9));
```

Comment:
combination of input values for each parameter needs to be considered. each parameter has around 5 values , 2 boundary values and 3 before and after boundary values.
The test cases need to be strengthened with combination of these values for each parameter.